.github/workflows/check_pr_titles.yaml

```
@@ -0,0 +1,118 @@
+name: Check PR titles
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - edited
+
+# Limit permissions for GITHUB_TOKEN for the entire workflow
+permissions:
+  contents: read
+  pull-requests: write  # Only allow PR comments/labels
+  # All other permissions are implicitly 'none'
+
+jobs:
+  check_title:
+    if: github.repository == 'leanprover-community/mathlib4' && (github.event.action == 'opened' || github.event.changes.title.from)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          ref: master
```

```suggestion
          ref: master
```
should this point to the branch with the PR?